### PR TITLE
Minor telemetry cleanup

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Refs #1323.
Refs #1443.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/util/telemetry/reporter.go [hunk 1]
```diff
@@ -78,18 +78,16 @@ type NewReporterOpts struct {
...
-		if err = opts.P.Update(func(s *state.State) { s.TelemetryLocked = true }); err != nil {
-			return nil, err
-		}
-	}
-
-	if err = opts.P.Update(func(s *state.State) { s.Telemetry = t }); err != nil {
+	err = opts.P.Update(func(s *state.State) {
+		s.Telemetry = t
+		s.TelemetryLocked = locked
+	})
```
Comment (AlekSi) [inline internal/util/telemetry/reporter.go:89]: No need to update state twice and trigger two updates

File: internal/util/telemetry/reporter_test.go [hunk 2]
```diff
@@ -30,37 +30,43 @@ func TestNewReporterLock(t *testing.T) {
...
-			expectedLock: true,
+			f:      &Flag{v: pointer.ToBool(false)},
+			t:      pointer.ToBool(false),
+			locked: true,
 		},
-		"DoNotTouch": {
-			f:            new(Flag),
-			dnt:          "enable",
-			expectedLock: true,
+		"DoNotTrack": {
```
Comment (AlekSi) [inline internal/util/telemetry/reporter_test.go:52]: That one was fun :)

File: internal/util/telemetry/telemetry.go [hunk 3]
```diff
@@ -64,14 +64,14 @@ func (s *Flag) UnmarshalText(text []byte) error {
 // The second returned value is true if the telemetry state should be locked, because of
 // setting telemetry via a command-line flag, an environment variable, or a filename.
 func initialState(f *Flag, dnt string, execName string, prev *bool, l *zap.Logger) (*bool, bool, error) {
-	var disable, locked bool
+	var disable bool
```
Comment (AlekSi) [inline internal/util/telemetry/telemetry.go:67]: I think mixing verbs and states makes is harder to understand

**Review by noisersup — COMMENTED** [review]
Thanks for improvements! :slightly_smiling_face: From this point I can see that there were couple of places when the code from last PR could be more consistent to the whole codebase. I just left one comment to discuss

File: internal/util/telemetry/telemetry.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (noisersup) [reply internal/util/telemetry/telemetry.go:67]: https://github.com/FerretDB/FerretDB/pull/1380/files#r1021010806

File: internal/util/telemetry/telemetry.go [hunk 4]
```diff
@@ -63,62 +63,66 @@ func (s *Flag) UnmarshalText(text []byte) error {
 //
 // The second returned value is true if the telemetry state should be locked, because of
 // setting telemetry via a command-line flag, an environment variable, or a filename.
-func initialState(f *Flag, dnt string, execName string, prev *bool, l *zap.Logger) (*bool, bool, error) {
-	var disable bool
-
+func initialState(f *Flag, dnt string, execName string, prev *bool, l *zap.Logger) (state *bool, locked bool, err error) {
```
Comment (noisersup) [inline internal/util/telemetry/telemetry.go:66]: :100:

**Review by noisersup — DISMISSED** [review]

**Review by w84thesun — APPROVED** [review]

**Review by noisersup — APPROVED** [review]

**Review by chilagrow — APPROVED** [review]

**Pull request merged**